src/ts/utils/roam.ts

```diff
@@ -68,69 +67,131 @@ export const Roam = {
+        const thisBlock = this.getRoamBlockInput() as HTMLElement;
+        if (this.hasChildren()) {
+            return this.activateBlock(DOM.getFirstChild(thisBlock))
+        }
+        return null;
+    },
+    async goToLastChild() {
+        const thisBlock = this.getRoamBlockInput() as HTMLElement;
+        if (this.hasChildren()) {
+            return this.activateBlock(DOM.getLastChild(thisBlock))
         }
+        return null;
+    },
+    isEmpty() {
+        return !this.getActiveRoamNode()?.text;
+    },
+    async addPlaceholder() {
+        if (this.isEmpty()) await this.writeText(' ');
+        await this.moveCursorToEnd();
+        return this.getRoamBlockInput()?.id as string;
+    },
+    async removePlaceholder(id: string) {
+        const currentId = this.getRoamBlockInput()?.id as string;
+        const placeholderElement = document.getElementById(id) as HTMLElement;
+        await this.activateBlock(placeholderElement)
+        await this.writeText('');
+        const currentElement = document.getElementById(currentId) as HTMLElement;
+        await this.activateBlock(currentElement);
+    },
+    async usePlaceholder(fn: ()=>void) {
+        let placeholderId = '';
+        if(this.isEmpty()){
+            placeholderId = await this.addPlaceholder();
+        }
+
+        await fn();
+
+        if (placeholderId) await this.removePlaceholder(placeholderId);
+    },
+
+    async createSiblingAbove(text?: string) {
+        await this.usePlaceholder(async ()=> {
+            await this.moveCursorToStart();
+            await Keyboard.pressEnter();
+        })
+        if (text) await this.writeText(text);
     },
     
-    async createSiblingBelow() {
-        this.moveCursorToEnd();
-        await Keyboard.pressEnter();
+    async createSiblingBelow(text?: string) {
+        await this.createFirstChild(text);
         await Keyboard.pressShiftTab(Keyboard.BASE_DELAY);
     },
 
-    async createFirstChild() {
-        this.moveCursorToEnd();
-        await Keyboard.pressEnter();
-        await Keyboard.pressTab();
+    async createFirstChild(text?: string) {
+        await this.moveCursorToEnd();
+        if (this.hasChildren()) await Keyboard.pressEnter();
+        else 
+            await this.usePlaceholder( async () => {
+                await Keyboard.pressEnter();
+                await Keyboard.pressTab();
+            })
+        if (text) await this.writeText(text);
     },
 
-    async createLastChild() {
-        await this.createSiblingBelow();
+    async createLastChild(text?: string) {
+        await this.createSiblingBelow(text);
         await Keyboard.pressTab();
     },
-
-    async createDeepestLastDescendant() {
+    
+    async createDeepestLastDescendant(text?: string) {
         await this.selectBlock();
         await Keyboard.simulateKey(Keyboard.RIGHT_ARROW);
-        await Keyboard.pressEnter();
+        await this.createSiblingBelow();
+        if (text) await this.writeText(text);
```

Comment:
I think you should move the check for text validity inside the `writeText` function instead of always gating it